code/basic.js

```diff
@@ -0,0 +1,485 @@
+  }
+}
+
+const setName = (name) => {
+  console.log('')
+  kitten.name = name
+  showMessage(`Oh wow! ${kitten.name}, that is a splendid name!`, 'bot')
+  currentQuestion++
+  fifthQuestion()
+}
+
+// This one is doin way too much work, need to be refactored?
+const handleResponse = (receivedMessage) => {
+  if (receivedMessage != '' && receivedMessage != ' ') {
+    console.log(kitten)
+    console.log('handleResponse, currentQuestion: ' + currentQuestion)
+    input.value = ''
+    // 1
+    if (currentQuestion === 1) {
+      if (receivedMessage.toLowerCase() != 'kittenbot go!') {
+        currentQuestion++
+        secondQuestion()
+      } else {
+        handleError()
+      }
+    // 2  
+    } else if (currentQuestion === 2) {
+      kitten.ownerName = receivedMessage; 
+      showMessage(receivedMessage, 'user') 
+      showMessage(`Well hello ${kitten.ownerName} it is very nice to meet you!`)
+      currentQuestion++
+      thirdQuestion()
+    // 3
+    } else if (currentQuestion === 3) {
+      showMessage(receivedMessage, 'user')    
+      if (receivedMessage.toLowerCase() === 'boy' || receivedMessage.toLowerCase() === 'girl' || receivedMessage.toLowerCase() === "doesn't matter") {
+        kitten.gender = receivedMessage
+        currentQuestion++
+        fourthQuestion()
+      } else {
+        handleError()
+      }
+
+    // 4
+    } else if (currentQuestion === 4) {
+      showMessage(receivedMessage, 'user')  
+      
+      //Success!
+      if (receivedMessage.length > 1 && kitten.nameConfirmation === false) {
+        console.log('')
+        setName(receivedMessage)
+      
+      // User asks for name suggestion
+      } else if (receivedMessage.length === 1) {
+        console.log('')
+        kitten.nameFirstLetter = receivedMessage
+        kitten.nameSuggestion = setRandomName(kitten.nameFirstLetter)
+      
+        // Bot found name starting with requested letter
+        if (setRandomNameSuccess) {
+          console.log('')
+          showMessage(`Hmm... something beginning with ${receivedMessage.toUpperCase()}... Ok how about we call ${kitten.personalPronoun} ${kitten.nameSuggestion}?`, 'bot')
+          userConfirmName()
+
+          // Bot could not find name starting with requested letter
+        } else {
+          console.log('')
+          showMessage(`Sorry, i don't know any names beginning with ${receivedMessage.toUpperCase()} :( ... is it ok if we just call ${kitten.personalPronoun} ${kitten.nameSuggestion}?`, 'bot')
+          userConfirmName()
+        } 
+      } else if (receivedMessage.toLowerCase() === 'ok' && kitten.nameConfirmation) {
+        console.log('')
+        setName(kitten.nameSuggestion)
+      } else if (receivedMessage === 'Hmm... not loving it' && kitten.nameConfirmation) {
+        console.log('')
+        kitten.nameSuggestion = setRandomName(kitten.nameFirstLetter)
+        showMessage(`How about ${kitten.nameSuggestion}?`, 'bot')
+        userConfirmName()
+      }
```

Comment:
It is a lot of `if` and `else`, maybe it would be easier to use elegant `switch`? https://www.w3schools.com/js/js_switch.asp